src/sidebar/store/modules/annotations.js

```diff
@@ -331,6 +332,48 @@ function hideAnnotation(id) {
   };
 }
 
+/**
+ * De-select and remove an annotation.
+ */
+function removeAndDeselectAnnotation(annotation) {
```

Comment:
My first thought is, looking at the code where this is actually called it appears to always create a list of 1 annotation so I'm not sure there's a need for a plural version anyway. 

Since this is dispatching a singular action inside the plural store function does this really have the advantage of being more efficient? Based on what you describe above it seems like it would still be calling every store subscriber for every dispatched action anyway although I see your point if it was a plural action that didn't dispatch other singular actions then it would be more efficient.